### DISCUSSION TITLE
Is a Docker action required to have DockerFile in the root?

### DISCUSSION THREAD

**Discussion by vidavidorra** [community/community#25195] [Actions]
<p>Is it required to have the <code>DockerFile</code> in the root of the repo?</p>
<p>I’m creating my first action and was following the <a href="https://help.github.com/en/actions/building-actions/creating-a-docker-container-action#create-a-dockerfile" rel="nofollow noopener">docs</a>, which doesn’t explicitly state that this is required. Is that the case, as long as <code>runs.image</code> in <code>action.yml</code> points to the file.</p>
<p>Also, perhaps more importantly, does the filename need to be case sensitive (i.e. <code>DockerFile</code> is okay, but <code>dockerfile</code> is not)?<br>
Reason I’m asking that is because I want to test the action in a workflow in the same repo. That is to verify the action works when I pused code changes. For reference, <a href="https://github.com/vidavidorra/github-action-renovate" rel="nofollow noopener">this is my repo</a> for this. I’ve had a build with <code>DockerFile</code>, but that didn’t run because the Docker repository name must be lowercase. See <a href="https://github.com/vidavidorra/github-action-renovate/runs/517627967?check_suite_focus=true" rel="nofollow noopener">this build</a>.<br>
After that I’ve moved to lowercase and the container ran (failue in the build is unrelated to this specific issue). See <a href="https://github.com/vidavidorra/github-action-renovate/runs/517638110?check_suite_focus=true" rel="nofollow noopener">this build</a>.</p>
<p>If there is a better/different way to test the actions locally, that is in the same repo, I’d love to hear that too. I’m new to actions but couldn’t find this specific information or practices in some other OSS actions I’ve looked into to debug this.</p>
<p>Thanks!</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/vidavidorra">@vidavidorra</a> ,</p>
<p>A Dockerfile is a text file that Docker reads in from top to bottom. It contains a bunch of instructions which informs Docker HOW the Docker image should get built. The Dockerfile is essentially the build instructions to build the image, you can say a Dockerfile is a recipe (or blueprint if that helps) for building Docker images.</p>
<p>From the description of Dockerfile, I think a Dockerfile is required for a Docker action. And we generally use “Dockerfile” to name it, this is the most recommended name.</p>
<p>In addition, if you setup the action in its own repository, when you wants to use this action the workflow, you also can directly use the syntax " <strong>uses: {owner}/{repo}@{ref}</strong>". For example “uses: vidavidorra/github-action-renovate@master” in your case.</p>

**Comment by vidavidorra** [comment, marked as answer]
<p>Thanks for the response. I probably wasn’t too clear about it in my initial message but I know how Docker works. Regardless, thanks for your clear explanation.</p>
<p>I’ve tried a whole bunch of different options, but apparently the name of the Dockerfile cannot contain any upercase characters other that the first uppercase character. So “Dockerfile” and “dockerfile” are both fine but “DockerFile” isn’t. That seemed to have been the issue as I now have successfull build (<a href="https://github.com/vidavidorra/github-action-renovate/runs/520538258?check_suite_focus=true" rel="nofollow noopener">example</a>).</p>
<p>I’ve also seen people use that, but that would only work for running the action on push of the master branch, and not for PRs.</p>
<pre><code>uses: vidavidorra/github-action-renovate@master
</code></pre>
<p>I’m now using the following with my Dockerfile in a <a href="https://github.com/vidavidorra/github-action-renovate/tree/cd4dee329720030349b278460f2f6ff8ddaca4c4/src" rel="nofollow noopener">src directory</a>. This should work for all branches and PRs as well.</p>
<pre><code>used: ./
</code></pre>
